Source/inv.cpp

```diff
@@ -836,19 +794,14 @@ int SwapItem(ItemStruct *a, ItemStruct *b)
 
 void CheckInvPaste(int pnum, int mx, int my)
 {
-	int r, sx, sy;
-	int i, j, xx, yy, ii;
-	bool done, done2h;
-	int il, cn, it, iv, ig, gt;
-	ItemStruct tempitem;
-
 	SetICursor(plr[pnum].HoldItem._iCurs + CURSOR_FIRSTITEM);
-	i = mx + (icursW / 2);
-	j = my + (icursH / 2);
-	sx = icursW28;
-	sy = icursH28;
-	done = false;
-	for (r = 0; (DWORD)r < NUM_XY_SLOTS && !done; r++) {
+	int i = mx + (icursW / 2);
+	int j = my + (icursH / 2);
+	int sx = icursW28;
+	int sy = icursH28;
+	bool done = false;
+	int r = 0;
+	for (; (DWORD)r < NUM_XY_SLOTS && !done; r++) {

```

Comment:
The (DWORD) is bonkers. None of the values involved is a DWORD. And it's a C style cast.